interview-questions.md

```diff
@@ -8,44 +8,61 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Researched answer:
+faves = { cereal1: "Fruity Pebbles", cereal2: "Life", cereal3: "Grape Nuts"}
+
+Researched answer: A hash in Ruby is an enumerable that means you can uses iterable methods on a hash. The default value of hashes is nil.
 
 2. What is a gem?
 
-Your answer:
+Your answer:  Ruby gems are package managers that help tie everything together as in Ruby on Rails. Gems are like little programs of Ruby code that contribute to the functionality of Ruby. 
 
-Researched answer:
+Researched answer: A gem is an open-source library of code. Gems contain code, documentation, and gemspec. A bundler and rails are examples of gems. 
 
 3. What is Ruby on Rails?
 
-Your answer:
+Your answer: Ruby on Rails is a complex system that allows Ruby to interact with databases as sort of a translator. A full-stack frame-work for both front-end and back-end development.
+
+Researched answer: Ruby on Rails uses the Model-View-Controller (MVC) architectural pattern used by many other web frameworks — one of the most well-known patterns in development. The MVC pattern separates a web application's code into three interconnected parts:
 
-Researched answer:
+The Model, which holds the data structure of the application.
+The View, which represents the visual part of the web app as web page templates.
+The Controller, which connects the data to the View and contains the business logic of the application.
 
 4. What is a relational database? Are there other kinds of databases?
 
-Your answer:
+Your answer: A relational database is a database that organizes data in the form of tables as it relates to other data within the database. 
+
+This is just one kind of database of many. 
 
-Researched answer:
+Researched answer: In relational databases, data structures remains separate from the physical storage structure. 
+
+Other forms of databases include: Centralised database,Distributed database, Personal database, End-user database, Commercial database, NoSQL database, Operational database, Relational database, Cloud database, Object-oriented database, Graph database.
 
 5. What are primary keys? Why are they important?
 
-Your answer:
+Your answer: Primary keys are the unique identifies in a data base that allow each instance created to be absolutely unique from one another despite whether or not the information contained in the unique instance is similar. Primary keys are incredibly important in data bases as it is very common to have for example, people with the same name and need a way to distinguish them from each other. 
 
-Researched answer:
+Researched answer: A primary key, also called a primary keyword, is a key in a relational database that is unique for each record. A primary key is necessary to ensure that a table record can always be uniquely identified. A primary key’s main features are:
+It must contain a unique value for each row of data.
+It cannot contain null values.
+Every row must have a primary key value.
```

Comment:
Nice work on this section. Your answers were great and you did an awesome job on the research.  👏